### PR TITLE
Fix a false positive and incorrect autocorrect for `RSpec/Capybara/SpecificActions`, `RSpec/Capybara/SpecificFinders` and `RSpec/Capybara/SpecificMatcher`

### PR DESCRIPTION
**Pull request by ydah**
Fix: https://github.com/rubocop/rubocop-capybara/issues/4

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [-] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: spec/rubocop/cop/rspec/capybara/specific_finders_spec.rb [hunk 1]
```diff
@@ -66,14 +147,32 @@
       ^^^^^^^^^^^^^^^^^^^^ Prefer `find_by` over `find`.
       find('[visible][id=some-id]')
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `find_by` over `find`.
-      find('[id=some-id][class=some-cls][focused]')
-      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `find_by` over `find`.
+      find('[id=some-id][visible][focused]')
```
Comment (Tietew) [inline spec/rubocop/cop/rspec/capybara/specific_finders_spec.rb:150]: CSS selector `[visible]` means the element has an attribute named `visible` (e.g. `<span visible="yes">`), NOT "the element is actually visible". CSS does not have `:visible` pseudo-class ---  it's a jQuery's extension.
And also, `[focused]` means the element has an attribute named `focused`, NOT "the element is now focused". `:focus` pseudo-class should be used instead.

Comment (ydah) [reply spec/rubocop/cop/rspec/capybara/specific_finders_spec.rb:150]: That is indeed true. I guess I was mistaken. Thanks for pointing that out to me.

**Comment by ydah** [comment]
I updated this PR.

**Comment by pirj** [comment]
I can't tell much about those cops, just that there were no errors:
```
$ rubocop --only RSpec/Capybara/MatchStyle --only RSpec/Capybara/SpecificActions --only RSpec/Capybara/SpecificFinders --only RSpec/Capybara/SpecificMatcher ../real-world-rspec
...
57497 files inspected, 63 offenses detected
```

Thank you!

**Comment by pirj** [comment]
```
/Users/pirj/source/real-world-rspec/solidus/frontend/spec/features/checkout_spec.rb:359:14: C: [Correctable] RSpec/Capybara/SpecificFinders: Prefer find_by over find.
      expect(find("#use_existing_card_yes")).to be_checked
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

57497 files inspected, 336 offenses detected, 336 offenses autocorrectable
```
for the other

**Comment by pirj** [comment]
```
/Users/pirj/source/real-world-rspec/publify/publify_core/spec/features/drafting_articles_spec.rb:25:5: C: RSpec/Capybara/SpecificActions: Prefer click_link over find('a').click.
    find('a[href="/admin/articles/1/edit"]').click
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

57497 files inspected, 63 offenses detected
```

**Comment by ydah** [comment]
I rebased this PR.

File: lib/rubocop/cop/rspec/mixin/css_selector.rb [hunk 2]
```diff
@@ -87,21 +57,11 @@ def attribute?(selector)
         def attributes(selector)
           selector.scan(/\[(.*?)\]/).flatten.to_h do |attr|
             key, value = attr.split('=')
+            value << ']' if value&.include?('[')
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/mixin/css_selector.rb:60]: This does not look right. Probably a shortcoming of us using regular expressions to parse CSS :-)

**Review by bquorning — COMMENTED** [review]
I would really like to see some direct spec coverage of the mixins, especially lib/rubocop/cop/rspec/mixin/css_selector.rb. For a start, it could just test the mentioned `@example` lines of each method.

**Comment by pirj** [comment]
Please don't merge this just yet, I'm extracting `rspec-capybara`.

**Comment by pirj** [comment]
May I kindly ask to re-target this to `rubocop-capybara`, @ydah ?

**Comment by ydah** [comment]
@pirj OK, I'll do it 👍 It may take a little time as we add mixin testing as well.

**Comment by ydah** [comment]
We are now open and PR here is closed.
https://github.com/rubocop/rubocop-capybara/pull/10

**Pull request closed without merging**